alf/utils/plot_tb_curves.py

```diff
@@ -27,23 +27,83 @@
+    """
+    CI_Z = {"80": 1.282, "85": 1.440, "90": 1.645, "95": 1.960, "99": 2.576}
+
+    def _get_ci_z(interval_mode):
+        """Get the corresponding Z value used in confidence interval computation."""
+        # mode -> "CI_X" where X is the percentage
+        ci_level = interval_mode.split("_")[1]
+        assert ci_level in CI_Z, "Invalid level value %s!" % ci_level
+        return CI_Z[ci_level]
+
+    y = np.array(list(map(np.mean, zip(*ys))))
+    std = np.array(list(map(np.std, zip(*ys))))
+    if interval_mode == "std":
+        min_y, max_y = y - std, y + std
+    elif interval_mode == "minmax":
+        min_y = np.array(list(map(np.min, zip(*ys))))
+        max_y = np.array(list(map(np.max, zip(*ys))))
+    elif interval_mode.startswith("CI_"):
+        z = _get_ci_z(interval_mode)
+        margin_err = std / np.sqrt(len(ys)) * z
+        min_y, max_y = y - margin_err, y + margin_err
+    else:
+        raise ValueError("Invalid interval mode: %s" % interval_mode)
+
+    return y, min_y, max_y
 
 
-# A namedtuple that represents a curve with mean and upper/lower bounds
 class MeanCurve(
         namedtuple(
-            "MeanCurve", ['x', 'y', 'min_y', 'max_y', 'name'],
-            default_value=())):
-    def average_y(self):
-        """Return the averaged ``y``, ``min_y``, and ``max_y`` over the x-axis.
+            "MeanCurve",
+            ['x', 'y', 'min_y', 'max_y', 'ay', 'min_ay', 'max_ay', 'name'],
+            default_value=None)):
+    @classmethod
+    def from_curves(cls, x, ys, interval_mode="std", name="MeanCurve"):
+        """Compute various curve statistics from a set of individual curves ``ys``
+        and a common ``x``, and create a class instance.
+
+        Args:
+            x (np.array): x steps
+            ys (list[np.array]): a list of curves
+            interval_mode (str): mode for computing error margin around the mean
+                y curve. Should be one of the four options: 1) "std", 2) "minmax",
+                and 3) "CI_X". The last one means confidence interval, where 'X'
+                should be one of ``(80,85,90,95,99)`` indicating the confidence
+                level (percentage).
+            name (str):
         """
-        return self.final_y(N=len(self.y))
+        # mean curve, lower and upper curve
+        y, min_y, max_y = _compute_y_interval(interval_mode, ys)
+        ays = [np.mean(y, keepdims=True) for y in ys]
+        # mean average_y, lower and upper average_y
+        # average_y can be used to indicate the changing trend of y
+        ay, min_ay, max_ay = map(lambda z: z.squeeze(-1),
+                                 _compute_y_interval(interval_mode, ays))
+        return cls(
+            x=x,
+            y=y,
+            min_y=min_y,
+            max_y=max_y,
+            ay=ay,
+            min_ay=min_ay,
+            max_ay=max_ay,
+            name=name)
 
     def final_y(self, N=1):
```

Comment:
Seems that this function is not used anywhere.

Reply:
> Seems that this function is not used anywhere.

I used this in my plotting scripts.